### ISSUE TITLE
Pulsar does not find discovered datasets

### ISSUE THREAD

**Issue by simonbray** [galaxyproject/pulsar#212]
I am trying to run the `gmx_sim` tool over pulsar on usegalaxy.eu. One of the outputs is a collection and should contain two files, but instead is empty. When I rerun the job without sending it to the pulsar server the collection contains two files (the expected behaviour).

The collection should contain datasets that match: `<discover_datasets pattern="(?P&lt;designation&gt;.*)\.xvg" visible="true" directory="." />`. Link: https://github.com/galaxycomputationalchemistry/galaxy-tools-compchem/blob/master/tools/gromacs/sim.xml#L247

I spoke to @gmauro and it seems pulsar will only return files which match one of these expected outputs - i.e., not those in the collection.

```      
"work_dir_outputs": [
        [
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/working/verbose.txt",
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/galaxy_dataset_13927188.dat"
        ],
        [
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/working/outp.tpr",
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/galaxy_dataset_13927187.dat"
        ],
        [
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/working/outp.edr",
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/galaxy_dataset_13927186.dat"
        ],
        [
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/working/outp.xtc",
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/galaxy_dataset_13927185.dat"
        ],
        [
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/working/outp.gro",
          "/data/dnb02/galaxy_db/job_working_directory/006/652/6652952/galaxy_dataset_13927184.dat"
        ]
      ],
```

It seems this could be a general issue affecting collections, hence the issue here.

ping @bgruening, @jmchilton

**Comment by bgruening** [comment]
Can confirm it with a separate tool.

**Comment by Slugger70** [comment]
Same for me with fastp. When running on pulsar with a list:paired input returns a collection with placeholder files but the files are empty.

**Comment by jmchilton** [comment]
Yes - there are TODOs in the code about this but I can't find another issue. Pulsar only collects the default Galaxy discovered dataset pattern. This needs to be parameterized and sent to Galaxy with the job definition. There are a lot of framework tests that fail on this - e.g. ``multi_output_configured``

**Comment by natefoo** [comment]
@jmchilton was this fully fixed by galxayproject/galaxy#12260? Can we close this?

**Comment by mvdbeek** [comment]
I think we can, yeah
